kafka-agent/src/test/java/io.strimzi.kafka.agent/KafkaAgentTest.java

```diff
@@ -0,0 +1,109 @@
+    private Server server;
+    private ContextHandler context;
+    private HttpRequest req;
+
+    @Before
+    public void setUp() throws URISyntaxException {
+        server = new Server();
+        ServerConnector conn = new ServerConnector(server);
+        conn.setPort(8080);
+        server.setConnectors(new Connector[] {conn});
+        context = new ContextHandler("/");
+
+        req = HttpRequest.newBuilder()
+                .uri(new URI("http://localhost:8080/"))
+                .GET()
+                .build();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (server != null) {
+            server.stop();
+        }
+    }
+
+    @Test
+    public void testBrokerRunningState() throws Exception {
+        final Gauge brokerState = mock(Gauge.class);
+        when(brokerState.value()).thenReturn((byte) 3);
+        KafkaAgent agent = new KafkaAgent(brokerState, null, null);
+        context.setHandler(agent.getServerHandler());
+        server.setHandler(context);
+        server.start();
+
+        HttpResponse<String> response = HttpClient.newBuilder()
+                .build()
+                .send(req, HttpResponse.BodyHandlers.ofString());
+        assertEquals(response.statusCode(), HttpServletResponse.SC_OK);
+
+        String expectedResponse = "{\"brokerState\":3}";
+        assertEquals(expectedResponse, response.body());
+    }
+
+    @Test
+    public void testBrokerRecoveryState() throws Exception {
+        final Gauge brokerState = mock(Gauge.class);
+        when(brokerState.value()).thenReturn((byte) 2);
+
+        final Gauge remainingLogs = mock(Gauge.class);
+        when(remainingLogs.value()).thenReturn((byte) 10);
+
+        final Gauge remainingSegments = mock(Gauge.class);
+        when(remainingSegments.value()).thenReturn((byte) 100);
+
+        KafkaAgent agent = new KafkaAgent(brokerState, remainingLogs, remainingSegments);
+        context.setHandler(agent.getServerHandler());
+        server.setHandler(context);
+        server.start();
+
+        HttpResponse<String> response = HttpClient.newBuilder()
+                .build()
+                .send(req, HttpResponse.BodyHandlers.ofString());
+        assertEquals(response.statusCode(), HttpServletResponse.SC_OK);
+
+        String expectedResponse = "{\"brokerState\":2,\"recoveryState\":{\"remainingLogsToRecover\":10,\"remainingSegmentsToRecover\":100}}";
+        assertEquals(expectedResponse, response.body());
+    }
+
+    @Test
+    public void testBrokerMetricNotFound() throws Exception {
+        KafkaAgent agent = new KafkaAgent(null, null, null);
+        context.setHandler(agent.getServerHandler());
+        server.setHandler(context);
+        server.start();
+
+        HttpResponse<String> response = HttpClient.newBuilder()
+                .build()
+                .send(req, HttpResponse.BodyHandlers.ofString());
+        assertEquals(response.statusCode(), HttpServletResponse.SC_NOT_FOUND);
```

Comment:
The expected value should be the first argument, and the actual value the 2nd. It doesn't matter when the assertion succeeds, but it results in a confusing message if the assertion fails.